app/models/agents/http_status_agent.rb

```diff
@@ -82,7 +82,8 @@ def check_this_url(url, local_headers)
 
       # Deal with failures
       if measured_result.result
-        payload.merge!({ 'response_received' => true, 'status' => current_status })
+        final_url = boolify(interpolated['disable_redirect_follow']) ? url : measured_result.result.to_hash[:url]
```

Comment:
This should have said `measured_result.result.env.url.to_s`.  The `url` here is a URI object, and `url` is a string, so `url != final_url` which is set to `redirected` always evaluates to true.  The specs are not actually testing anything because `agent.faraday` is mocked to return what they want.